src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+
+from hidra import Transfer
+from plugins.asapo_producer import AsapoWorker
+
+logger = logging.getLogger(__name__)
+
+
+class AsapoTransfer:
+    def __init__(self, endpoint, beamtime, token, n_threads, file_regex,
+                 default_data_source=None, timeout=5, beamline='auto',
+                 start_file_idx=1):
+
+        self.signal_host = "localhost"
+        self.targets = [["localhost", "50101", 1]]
+        self.asapo_worker = AsapoWorker(endpoint, beamtime, token, n_threads, file_regex,
+                                        default_data_source, timeout, beamline,
+                                        start_file_idx)
+
+        self.start_transfer()
+
+    def start_transfer(self):
+
+        query = Transfer("STREAM_METADATA", self.signal_host)
+        query.initiate(self.targets)
+        query.start()
+        self.run(query)
+        query.stop()
+
+    def run(self, query):
+        while True:
+            try:
+                [metadata, data] = query.get()
+            except:
+                break
+
+            if metadata is not None and data is not None:
+                local_path = f'{metadata["relative_path"]}/{metadata["filename"]}'
+                self.asapo_worker.send_message(local_path, metadata)
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                     description='Transfer files metadata to ASAPO')
+    parser.add_argument('--endpoint', type=str, help='ASAPO produces endpoint',
+                        default='localhost:8400')
+    parser.add_argument('--beamtime', type=str,
+                        help='ASAPO produces beamtime',
+                        default='asapo_test')
+    parser.add_argument('--beamline', type=str,
+                        help='ASAPO produces beamline',
+                        default='auto')
+    parser.add_argument('--default-data-source', type=str, help='ASAPO data_source',
+                        default='test')
+    parser.add_argument('--token', type=str, help='ASAPO produces token',
+                        default='eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiJjMWY2OG0ydWlkZDE3dWxmaDN1ZyIsInN1YiI6ImJ0X2FzYXBvX3Rlc3QiLCJFeHRyYUNsYWltcyI6eyJBY2Nlc3NUeXBlcyI6WyJyZWFkIl19fQ.zo7ZDfY2sf4o9RYuXpxNR9kHLG594xr-SE5yLoyDC2Q')
```

Comment:
These should not have default arguments. It is better to get an error then a silently wrong result.

Reply:
Done.